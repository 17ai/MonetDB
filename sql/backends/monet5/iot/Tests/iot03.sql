-- Setting stream parameters after iot00
set schema iot;
set optimizer='iot_pipe';

-- Example of a window based action
create table result3(like stmp);

create procedure cq03()
begin
	call iot.window('iot','stmp',2);
	insert into result3 select min(t), count(*), avg(val) from stmp;
	call iot.tumble('iot','stmp',2);
end;
call iot.query('iot','cq03');

insert into stmp values('2005-09-23 12:34:26.000',1,11.0);
insert into stmp values('2005-09-23 12:34:27.000',1,11.0);
insert into stmp values('2005-09-23 12:34:28.000',1,13.0);
insert into stmp values('2005-09-23 12:34:28.000',1,13.0);

call iot.deactivate('iot','cq03');

select * from stmp;
select * from result3;

select * from  iot.queries();
