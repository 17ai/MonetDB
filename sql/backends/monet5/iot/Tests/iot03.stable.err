stderr of test 'iot03` in directory 'sql/backends/monet5/iot` itself:


# 08:44:01 >  
# 08:44:01 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=34418" "--set" "mapi_usock=/var/tmp/mtest-31022/.s.monetdb.34418" "--set" "monet_prompt=" "--forcemito" "--dbpath=/export/scratch1/mk/iot//Linux/var/MonetDB/mTests_sql_backends_monet5_iot"
# 08:44:01 >  

# builtin opt 	gdk_dbpath = /export/scratch1/mk/iot//Linux/var/monetdb5/dbfarm/demo
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_vmtrim = no
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 34418
# cmdline opt 	mapi_usock = /var/tmp/mtest-31022/.s.monetdb.34418
# cmdline opt 	monet_prompt = 
# cmdline opt 	gdk_dbpath = /export/scratch1/mk/iot//Linux/var/MonetDB/mTests_sql_backends_monet5_iot
# cmdline opt 	gdk_debug = 536870922

# 08:44:01 >  
# 08:44:01 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-31022" "--port=34418"
# 08:44:01 >  


# 08:44:01 >  
# 08:44:01 >  "Done."
# 08:44:01 >  

