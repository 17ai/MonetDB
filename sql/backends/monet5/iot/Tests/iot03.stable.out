stdout of test 'iot03` in directory 'sql/backends/monet5/iot` itself:


# 08:44:01 >  
# 08:44:01 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=34418" "--set" "mapi_usock=/var/tmp/mtest-31022/.s.monetdb.34418" "--set" "monet_prompt=" "--forcemito" "--dbpath=/export/scratch1/mk/iot//Linux/var/MonetDB/mTests_sql_backends_monet5_iot"
# 08:44:01 >  

# MonetDB 5 server v11.24.0
# This is an unreleased version
# Serving database 'mTests_sql_backends_monet5_iot', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 15.589 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2016 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.da.cwi.nl:34418/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-31022/.s.monetdb.34418
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
# MonetDB/iot loaded

Ready.

# 08:44:01 >  
# 08:44:01 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-31022" "--port=34418"
# 08:44:01 >  

#set schema iot;
#set optimizer='iot_pipe';
#create stream table stmp3 (t timestamp, sensor integer, val decimal(8,2)) ;
#create table result3(like stmp3);
#create procedure cq03()
#begin
#	call iot.window('iot','stmp3',2);
#	insert into result3 select min(t), count(*), avg(val) from stmp3;
#end;
#insert into stmp3 values('2005-09-23 12:34:26.000',1,9.0);
[ 1	]
#insert into stmp3 values('2005-09-23 12:34:27.000',1,11.0);
[ 1	]
#insert into stmp3 values('2005-09-23 12:34:28.000',1,13.0);
[ 1	]
#insert into stmp3 values('2005-09-23 12:34:28.000',1,15.0);
[ 1	]
#select * from stmp3;
% iot.stmp3,	iot.stmp3,	iot.stmp3 # table_name
% t,	sensor,	val # name
% timestamp,	int,	decimal # type
% 26,	1,	10 # length
#select * from result3;
% iot.result3,	iot.result3,	iot.result3 # table_name
% t,	sensor,	val # name
% timestamp,	int,	decimal # type
% 26,	1,	10 # length
[ 2005-09-23 12:34:26.000000,	2,	10.00	]
[ 2005-09-23 12:34:28.000000,	2,	14.00	]
#select * from iot.errors();
% .,	. # table_name
% table,	error # name
% clob,	clob # type
% 0,	0 # length
#drop procedure cq03;
#drop table stmp3;
#drop table result3;

# 08:44:01 >  
# 08:44:01 >  "Done."
# 08:44:01 >  

